crates/burn-core/src/nn/rnn/lstm.rs

```diff
@@ -81,25 +64,43 @@ impl LstmConfig {
 impl<B: Backend> Lstm<B> {
     /// Applies the forward pass on the input tensor. This LSTM implementation
     /// returns the cell state and hidden state for each element in a sequence (i.e., across `seq_length`),
-    /// producing 3-dimensional tensors where the dimensions represent [batch_size, sequence_length, hidden_size].
+    /// producing 3-dimensional tensors where the dimensions represent `[batch_size, sequence_length, hidden_size]`.
     ///
-    /// Parameters:
-    ///     batched_input: The input tensor of shape [batch_size, sequence_length, input_size].
-    ///     state: An optional tuple of tensors representing the initial cell state and hidden state.
-    ///            Each state tensor has shape [batch_size, hidden_size].
+    /// ## Parameters:
+    /// - batched_input: The input tensor of shape `[batch_size, sequence_length, input_size]`.
+    /// - state: An optional tuple of tensors representing the initial cell state and hidden state.
+    ///            Each state tensor has shape `[batch_size, hidden_size]`.
     ///            If no initial state is provided, these tensors are initialized to zeros.
     ///
-    /// Returns:
-    ///     A tuple of tensors, where the first tensor represents the cell states and
-    ///     the second tensor represents the hidden states for each sequence element.
-    ///     Both output tensors have the shape [batch_size, sequence_length, hidden_size].
+    /// ## Returns:
+    /// A tuple of tensors, where the first tensor represents the cell states and
+    /// the second tensor represents the hidden states for each sequence element.
+    /// Both output tensors have the shape `[batch_size, sequence_length, hidden_size]`.
     pub fn forward(
         &self,
         batched_input: Tensor<B, 3>,
         state: Option<(Tensor<B, 2>, Tensor<B, 2>)>,
     ) -> (Tensor<B, 3>, Tensor<B, 3>) {
-        let [batch_size, seq_length, _] = batched_input.shape().dims;
-        let device = &batched_input.device();
+        let device = batched_input.device();
+        let [batch_size, seq_length, _] = batched_input.dims();
+
+        self.forward_iter(
+            batched_input.iter_dim(1).zip(0..seq_length),
+            state,
+            batch_size,
+            seq_length,
+            &device,
+        )
+    }
+
+    fn forward_iter<I: Iterator<Item = (Tensor<B, 3>, usize)>>(
+        &self,
+        input_timestep_iter: I,
+        state: Option<(Tensor<B, 2>, Tensor<B, 2>)>,
+        batch_size: usize,
+        seq_length: usize,
+        device: &B::Device,
+    ) -> (Tensor<B, 3>, Tensor<B, 3>) {
         let mut batched_cell_state = Tensor::zeros([batch_size, seq_length, self.d_hidden], device);
         let mut batched_hidden_state =
             Tensor::zeros([batch_size, seq_length, self.d_hidden], device);
```

Comment:
Not linked to the current change, but is zero necessary or empty would be enough here?